frontend/javascripts/dashboard/dataset/dataset_settings_viewconfig_tab.tsx

```diff
@@ -14,24 +14,89 @@ import {
+import { getAgglomeratesForDatasetLayer, getMappingsForDatasetLayer } from "admin/admin_rest_api";
 
 const FormItem = Form.Item;
 
-export default function DatasetSettingsViewConfigTab() {
+export default function DatasetSettingsViewConfigTab(props: {
+  datasetId: APIDatasetId;
+  dataStoreURL: string | undefined;
+}) {
+  const { datasetId, dataStoreURL } = props;
+  const [availableMappingsPerLayerCache, setAvailableMappingsPerLayer] = useState(
+    {} as Record<string, [string[], string[]]>,
+  );
+
+  // biome-ignore lint/correctness/useExhaustiveDependencies: Update is not necessary when availableMappingsPerLayer[layerName] changes.
+  const validateDefaultMappings = useMemo(
+    () => async (configStr: string, dataStoreURL: string, datasetId: APIDatasetId) => {
+      let config = {} as DatasetConfiguration["layers"];
+      try {
+        config = JSON.parse(configStr);
+      } catch (e: any) {
+        return Promise.reject(new Error("Invalid JSON format for : " + e.message));
+      }
+      const layerNamesWithDefaultMappings = Object.keys(config).filter(
+        (layerName) => config[layerName].mapping != null,
+      );
+
+      const maybeMappingRequests = layerNamesWithDefaultMappings.map(async (layerName) => {
+        if (layerName in availableMappingsPerLayerCache) {
+          return availableMappingsPerLayerCache[layerName];
+        }
+        let jsonAndAgglomerateMappings = [[], []] as [string[], string[]];
+        try {
+          jsonAndAgglomerateMappings = await Promise.all([
+            getMappingsForDatasetLayer(dataStoreURL, datasetId, layerName),
+            getAgglomeratesForDatasetLayer(dataStoreURL, datasetId, layerName),
+          ]);
+        } catch (e: any) {
+          console.error(e);
+          throw new Error(messages["mapping.loading_failed"](layerName));
+        }
+        setAvailableMappingsPerLayer((prev) => ({
+          ...prev,
+          [layerName]: jsonAndAgglomerateMappings,
+        }));
+        return jsonAndAgglomerateMappings;
+      });
+      const mappings = await Promise.all(maybeMappingRequests);
+      const errors = layerNamesWithDefaultMappings
+        .map((layerName, index) => {
+          const [mappingsForLayer, agglomeratesForLayer] = mappings[index];
+          const mappingType = config[layerName]?.mapping?.type;
+          const mappingName = config[layerName]?.mapping?.name;
+          const doesMappingExist =
+            mappingType === "HDF5"
+              ? agglomeratesForLayer.some((agglomerate) => agglomerate === mappingName)
+              : mappingsForLayer.some((mapping) => mapping === mappingName);
+          return doesMappingExist
+            ? null
+            : `The default mapping "${mappingName}" of type "${mappingType}" does not exist for layer ${layerName}.`;
+        })
+        .filter((error) => error != null);
+      if (errors.length > 0) {
+        throw new Error("The following default mappings are invalid: " + errors.join("\n"));
```

Comment:
Same here. Is this error even displayed somewhere?

Reply:
Yes, like the other errors / rejected promisses, this message is shown below the layer config input in the view configurations

Reply:
![image](https://github.com/scalableminds/webknossos/assets/39529669/e3f91663-0450-4c1d-b55b-232b96650a75)
(Note that I removed the "default" from the error message as the property name of the setting no longer contains default in its name. In a prior version of this branch the property was called `defaultMapping` instead of just `mapping`)